t4-devkit/t4_devkit/common/geometry.py

```diff
@@ -51,15 +51,15 @@ def view_points(
         # distortion is [k1, k2, p1, p2, k3, k4, k5, k6, s1, s2, s3, s4]
         while len(D) < 12:
             D = np.insert(D, len(D), 0)
-        x_ = points[0] / points[2]
-        y_ = points[1] / points[2]
+        k1, k2, p1, p2, k3, k4, k5, k6, s1, s2, s3, s4 = D
+
+        x_ = points[0]
+        y_ = points[1]
         r2 = x_**2 + y_**2
-        f1 = (1 + D[0] * r2 + D[1] * r2**2 + D[4] * r2**3) / (
-            1 + D[5] * r2 + D[6] * r2**2 + D[7] * r2**3
-        )
+        f1 = (1 + k1 * r2 + k2 * r2**2 + k3 * r2**3) / (1 + k4 * r2 + k5 * r2**2 + k6 * r2**3)
         f2 = x_ * y_
-        x__ = x_ * f1 + 2 * D[2] * f2 + D[3] * (r2 + 2 * x_**2) + D[8] * r2 + D[9] * r2**2
-        y__ = y_ * f1 + D[2] * (r2 + 2 * y_**2) + 2 * D[3] * f2 + D[10] * r2 + D[11] * r2**2
+        x__ = x_ * f1 + 2 * p1 * f2 + p2 * (r2 + 2 * x_**2) + s1 * r2 + s2 * r2**2
+        y__ = y_ * f1 + p1 * (r2 + 2 * y_**2) + 2 * p2 * f2 + s3 * r2 + s4 * r2**2
         u = viewpad[0, 0] * x__ + viewpad[0, 2]
         v = viewpad[1, 1] * y__ + viewpad[1, 2]
         points = np.stack([u, v, np.ones(nbr_points)], axis=0)
```

Comment:
[Must] Need to change `np.ones` to `points[2]`

Reply:
Thanks, I've fixed in https://github.com/tier4/tier4_perception_dataset/pull/133/commits/65b7f9ca12b2a75e6829ed4a95df5259f9d7f5e1.